specs/WinRT/MakeMSIX-spec.md

```diff
@@ -0,0 +1,1002 @@
+                extractPackageOptions.OverwriteOutputFilesIfExists(true);
+                MakeMSIXManager::ExtractPackage(file.path().c_str(), outputDirForPackage.c_str(), extractPackageOptions).get();
+            }
+
+            std::filesystem::path outputDirRootForPackedPackages{ outputDirRoot };
+            outputDirRootForPackedPackages /= L"packedPackages";
+
+            // Iterate through packages and repack each one.
+            for (const auto& packageDir : std::filesystem::directory_iterator(outputDirRootForPackages))
+            {
+                std::filesystem::path outputPackagePath{ outputDirRootForPackedPackages };
+                std::wstring outputPackageFileName = packageDir.path().filename().wstring() + L".appx";
+                outputPackagePath /= outputPackageFileName;
+
+                // Change the version when re-packing
+                auto createPackageOptions = CreatePackageOptions();
+                createPackageOptions.OverwriteOutputFileIfExists(true);
+                createPackageOptions.Version(newVersion);
+                MakeMSIXManager::CreatePackage(packageDir.path().c_str(), outputPackagePath.c_str(), createPackageOptions).get();
+            }
+
+            std::filesystem::path outputPathForBundle{ outputDirRoot };
+            outputPathForBundle /= bundleFilePath.filename();
+            // Re-bundle with the new version
+            auto createBundleOptions = CreateBundleOptions();
+            createBundleOptions.OverwriteOutputFileIfExists(true);
+            createBundleOptions.Version(newVersion);
+            MakeMSIXManager::CreateBundle(outputDirRootForPackedPackages.c_str(), outputPathForBundle.c_str(), createBundleOptions).get();
+        }
+        catch (winrt::hresult_error const& ex)
+        {
+            OutputDebugString(ex.message().c_str());
+            winrt::check_hresult(ex.code());
+        }
+    }
+```
+
+## 2.3. Helper methods for examples
+
+Windows has existing non-WinRT APIs for signing packages available in the CryptoAPI
+https://learn.microsoft.com/en-us/windows/win32/seccrypto/using-cryptography
+For completeness purposes some helper methods to instead find and call signtool.exe
+from the Windows SDK are included here.
+```cpp
+    /// <summary>
+    /// Create a process and synchronously wait for it to exit.
+    /// </summary>
+    /// <param name="commandLine">CommandLine argument for CreateProcess</param>
+    /// <param name="exitCode">Exit code of the process</param>
+    /// <returns>Success if process is created and exit code is returned.</returns>
+    HRESULT CreateProcessAndWaitForExitCode(std::wstring commandLine, DWORD& exitCode)
+    {
+        STARTUPINFO startupInfo;
+        ZeroMemory(&startupInfo, sizeof(startupInfo));
+        startupInfo.cb = sizeof(startupInfo);
+
+        PROCESS_INFORMATION processInformation;
+        ZeroMemory(&processInformation, sizeof(processInformation));
+
+        if (!CreateProcess(
```

Comment:
C#ize :-)

FYI C# `p = new Process(); ... p.Start(); p.WaitForExit();` is the C# equivalent to CreateProcess, WaitForSingleHandle, etc